### PR TITLE
Add Bk engine

### PR DESCRIPTION
**Pull request by MichaelHinrichs**
### SteamDB app page links to a few games using this
https://steamdb.info/app/528060/
https://steamdb.info/app/1002290/
https://steamdb.info/app/1225780/
https://steamdb.info/app/370280/

### Brief explanation of the change
Add detection for BK Engine.

### CODE REVIEW

File: descriptions/Engine.BKEngine.md [hunk 1]
```diff
@@ -0,0 +1 @@
+[**BK Engine**](https://bke.bakery.moe/index_en.html) is a Cross Platform Visual Novel Game Engine. It is free, [**open source**](https://github.com/BKEngine/Creator), and supports Most Desktop and Mobile OS
```
Comment (Lifeismana) [inline descriptions/Engine.BKEngine.md:1]:
```suggestion
[**BK Engine**](https://bke.bakery.moe/index_en.html) is a Visual Novel Game Engine.
```

File: tests/types/Engine.BKEngine.txt
Comment (Lifeismana) [inline tests/types/Engine.BKEngine.txt:1]: add these strings and run the generate strings script
```
data_a.bkarc
PaperPlane.app/Contents/Resources/data_a.bkarc
```

**Review by Lifeismana — CHANGES_REQUESTED** [review]
add some non matching tests

File: descriptions/Engine.BKEngine.md (hunk 1 above)
Comment (MichaelHinrichs) [reply descriptions/Engine.BKEngine.md:1]: That barely tells anything. "It's an engine"? Yeah, so is everything else in the "Engine" category.

Comment (Lifeismana) [reply descriptions/Engine.BKEngine.md:1]: No it tells that it's a visual novel engine, which is enough, if someone wants to know more he'll just click on the link

**Pull request merged**